### PR TITLE
migrated do_verify_email component to typescript

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
migrated do_verify_email component to typescript
#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13715

#### Related Pull Requests

- Has server changes  - NO
- Has redux changes - NO
- Has mobile changes - NO

### CODE REVIEW

File: components/do_verify_email/do_verify_email.tsx [hunk 1]
```diff
@@ -1,104 +1,88 @@
...
-    constructor(props) {
+export default class DoVerifyEmail extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
         super(props);
 
         this.state = {
             verifyStatus: 'pending',
-            serverError: '',
+            serverError: null
+            ,
```
Comment (jespino) [inline components/do_verify_email/do_verify_email.tsx:49]: this trailing comma should be probably in the previous line.

File: components/do_verify_email/do_verify_email.tsx [hunk 2]
```diff
@@ -1,104 +1,88 @@
...
-            email_verified: PropTypes.bool,
-        }),
-
-        isLoggedIn: PropTypes.bool.isRequired,
-    }
+type Props = {
+    location: {
+        search: string;
+    };
+    siteName: string | undefined;
```
Comment (devinbinnie) [inline components/do_verify_email/do_verify_email.tsx:26]: You can use an optional parameter here as a shortcut for `string | undefined`:
```suggestion
    siteName?: string;
```

**Comment by lindy65** [comment]
@jespino @pradeepmurugesan - can you help with test steps for this PR please?  Is it testing the flow around email verification on sign-up / changing email address ?

**Comment by pradeepmurugesan** [comment]
> @jespino @pradeepmurugesan - can you help with test steps for this PR please? Is it testing the flow around email verification on sign-up / changing email address ?

Hi @lindy65 . I tested the following use case. 

1. Set the email verification to true in the server -> system console
2. Create a new account.
3. Click on the verification email.

I checked if the email is verified successfully. 

I am not sure if any other flow touches this component.

**Comment by lindy65** [comment]
Ah, perfect! Thanks @pradeepmurugesan - I will test using these steps and `@jespino` can weigh in if there are any other areas to test 👍

**Comment by jespino** [comment]
@pradeepmurugesan @lindy65 yes, I think the steps proposed sound sensible :)

**Pull request merged**